tools/traffic/workers/blob_status_tracker.go

```diff
@@ -0,0 +1,253 @@
+	waitGroup *sync.WaitGroup,
+	logger logging.Logger,
+	config *config.WorkerConfig,
+	keyChannel chan *UnconfirmedKey,
+	table *table.BlobStore,
+	disperser clients.DisperserClient,
+	generatorMetrics metrics.Metrics) BlobStatusTracker {
+
+	return BlobStatusTracker{
+		ctx:                               ctx,
+		waitGroup:                         waitGroup,
+		logger:                            logger,
+		config:                            config,
+		keyChannel:                        keyChannel,
+		confirmedBlobs:                    table,
+		disperser:                         disperser,
+		unconfirmedBlobs:                  make([]*UnconfirmedKey, 0),
+		blobsInFlightMetric:               generatorMetrics.NewGaugeMetric("blobs_in_flight"),
+		getStatusLatencyMetric:            generatorMetrics.NewLatencyMetric("get_status"),
+		confirmationLatencyMetric:         generatorMetrics.NewLatencyMetric("confirmation"),
+		getStatusErrorCountMetric:         generatorMetrics.NewCountMetric("get_status_ERROR"),
+		unknownCountMetric:                generatorMetrics.NewCountMetric("get_status_UNKNOWN"),
+		processingCountMetric:             generatorMetrics.NewCountMetric("get_status_PROCESSING"),
+		dispersingCountMetric:             generatorMetrics.NewCountMetric("get_status_DISPERSING"),
+		failedCountMetric:                 generatorMetrics.NewCountMetric("get_status_FAILED"),
+		insufficientSignaturesCountMetric: generatorMetrics.NewCountMetric("get_status_INSUFFICIENT_SIGNATURES"),
+		confirmedCountMetric:              generatorMetrics.NewCountMetric("get_status_CONFIRMED"),
+		finalizedCountMetric:              generatorMetrics.NewCountMetric("get_status_FINALIZED"),
+	}
+}
+
+// Start begins the status goroutine, which periodically polls
+// the disperser service to verify the status of blobs.
+func (verifier *BlobStatusTracker) Start() {
+	verifier.waitGroup.Add(1)
+	go verifier.monitor()
+}
+
+// monitor periodically polls the disperser service to verify the status of blobs.
+func (verifier *BlobStatusTracker) monitor() {
+	ticker := time.NewTicker(verifier.config.VerifierInterval)
+	for {
+		select {
+		case <-(*verifier.ctx).Done():
+			verifier.waitGroup.Done()
+			return
+		case key := <-verifier.keyChannel:
+			verifier.unconfirmedBlobs = append(verifier.unconfirmedBlobs, key)
+		case <-ticker.C:
+			verifier.poll()
+		}
+	}
+}
+
+// poll checks all unconfirmed keys to see if they have been confirmed by the disperser service.
+// If a Key is confirmed, it is added to the blob confirmedBlobs and removed from the list of unconfirmed keys.
+func (verifier *BlobStatusTracker) poll() {
+
+	// FUTURE WORK If the number of unconfirmed blobs is high and the time to confirm is high, this is not efficient.
+	// Revisit this method if there are performance problems.
+
+	nonFinalBlobs := make([]*UnconfirmedKey, 0)
+	for _, key := range verifier.unconfirmedBlobs {
+
+		blobStatus := verifier.getBlobStatus(key)
+		if blobStatus == nil {
```

Comment:
should probably log error as this case is unexpected

Reply:
added